src/Build.UnitTests/WarningsAsMessagesAndErrors_Tests.cs

```diff
@@ -21,6 +22,30 @@ public WarningsAsMessagesAndErrorsTests(ITestOutputHelper output)
             _output = output;
         }
 
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Netcoreapp, "https://github.com/microsoft/msbuild/issues/5158")]
+        public void TaskNodesDieAfterBuild()
+        {
+            using (TestEnvironment env = TestEnvironment.Create())
+            {
+                string introToPID = "PID to shut down is ";
+                string afterPID = "(EndPID)";
+                string pidTaskProject = $@"<Project>
+    <UsingTask TaskName=""ProcessIdTask"" AssemblyName=""Microsoft.Build.Engine.UnitTests"" TaskFactory=""TaskHostFactory"" />
+    <Target Name='AccessPID'>
+        <ProcessIdTask />
+    </Target>
+      </Project>";
+                TransientTestFile project = env.CreateFile("testProject.csproj", pidTaskProject);
+                MockLogger logger = new MockLogger(_output);
+                ObjectModelHelpers.BuildTempProjectFileWithTargetsExpectSuccess(project.Path, null, null, logger);
+                int start = logger.FullLog.IndexOf(introToPID) + introToPID.Length;
+                int length = logger.FullLog.IndexOf(afterPID) - start;
+                string pid = logger.FullLog.Substring(start, length);
```

Comment:
Instead of grepping through the logging, can you change the test to return the PID of its process as a property, capture the property into a named property when you call it, and read the property out of the built project instance? That's a much cleaner way to do things (and a good exercise for you in using our API).